backend/dataall/modules/worksheets/services/worksheet_query_result_service.py

```diff
@@ -0,0 +1,139 @@
+import csv
+import io
+import os
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from openpyxl import Workbook
+
+from dataall.base.aws.s3_client import S3_client
+from dataall.base.db import exceptions
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.modules.worksheets.db.worksheet_models import WorksheetQueryResult
+from dataall.modules.worksheets.db.worksheet_repositories import WorksheetRepository
+from dataall.modules.worksheets.services.worksheet_service import WorksheetService
+
+if TYPE_CHECKING:
+    try:
+        from sqlalchemy.orm import Session
+        from mypy_boto3_s3.client import S3Client
+    except ImportError:
+        print('skipping type checks as stubs are not installed')
+        S3Client = None
+        Session = None
+
+
+class WorksheetQueryResultService:
+    SupportedFormats = {'csv', 'xlsx'}
+
```

Comment:
As a convention, in data.all backed the `api` layer is responsible of the input validation of the API calls; in the `service` layer we verify permissions and execute business logic. Which means that in this case we should place lines 29-40 in `dataall/modules/worksheets/api/resolvers.py`

Reply:
done